hrms/payroll/doctype/salary_component/salary_component.py

```diff
@@ -32,3 +54,13 @@ def validate_abbr(self):
 			separator="_",
 			filters={"name": ["!=", self.name]},
 		)
+
+	@frappe.whitelist()
+	def update_salary_structures(self, field, value):
+		SalaryDetail = frappe.qb.DocType("Salary Detail")
+		SalaryStructure = frappe.qb.DocType("Salary Structure")
+		frappe.qb.update(SalaryDetail).inner_join(SalaryStructure).on(
+			SalaryDetail.parent == SalaryStructure.name
+		).set(SalaryDetail[field], value).where(
+			(SalaryDetail.salary_component == self.name) & (SalaryStructure.docstatus != 2)
+		).run()
```

Comment:
Although this is performant, let's work with doc objects to maintain version history so that people can revert to the original formula if needed. Check reference for maintaining history with direct `db_update` here:

https://github.com/frappe/erpnext/blob/b70f3de16be169a723842a3a99c046a3809d0768/erpnext/manufacturing/doctype/bom_update_log/bom_updation_utils.py#L28-L43